ts/screens/services/ServicesWebviewScreen.tsx

```diff
@@ -59,24 +94,40 @@ const ServicesWebviewScreen: React.FunctionComponent<Props> = (
     <BaseScreenComponent goBack={handleGoBack}>
       <SafeAreaView style={styles.flex}>
         <Content contentContainerStyle={styles.flex}>
-          {!cookieError && isCookieAvailable && (
-            <RegionServiceWebView uri={props.url} onModalClose={handleGoBack} />
-          )}
+          {!cookieError &&
+            isCookieAvailable &&
+            props.navigationParams.isRight() && (
+              <RegionServiceWebView
+                uri={props.navigationParams.value.url}
+                onModalClose={handleGoBack}
+              />
+            )}
         </Content>
       </SafeAreaView>
     </BaseScreenComponent>
   );
 };
 const mapStateToProps = (state: GlobalState) => {
-  const maybeUrl = fromNullable(internalRouteNavigationParamsSelector(state))
-    .chain(params => fromNullable(params.url))
-    .getOrElse("");
+  const params = ServicesWebviewParams.decode(
+    internalRouteNavigationParamsSelector(state)
+  );
 
-  // TODO Add TokenName parameter from service metadata
+  // Get TokenName parameter from service metadata
+  const servicesMetadataByID = servicesMetadataByIdSelector(state);
+
+  const tokenName = () => {
+    if (params.isRight()) {
+      const metadata = servicesMetadataByID[params.value.serviceID];
+      if (pot.isSome(metadata)) {
+        return metadata.value && metadata.value.token_name;
+      }
+    }
+    return undefined;
+  };
 
   return {
-    url: maybeUrl,
-    token: tokenFromNameSelector("walletToken")(state)
+    navigationParams: params,
+    token: tokenName() && tokenFromNameSelector(tokenName() as TokenName)(state)
```

Comment:
Perhaps some code could be refactored in a cleaner way
This is my attempt 751bb3dba45d9d48c195b7eabc9a480ef5a57cdf
Could it give an improvement?

Reply:
Yep sure! I couldn't figure out how to improve the part of `mapStateToProps` this is absolutely better!